consensus/safety-rules/src/tests/suite.rs

```diff
@@ -235,7 +231,7 @@ fn test_sign_proposal_with_bad_signer(safety_rules: &Callback) {
     let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer);
     safety_rules.sign_proposal(a1.block().block_data()).unwrap();
 
-    let bad_signer = ValidatorSigner::from_int(0xef);
+    let bad_signer = ValidatorSigner::random([0xfu8; 32]);
```

Comment:
why this change?

Reply:
any thoughts on this?

Reply:
See my response in https://reviewable.io/reviews/aptos-labs/aptos-core/2195#-N7o-awqg9zpxmKoEnh2-r3-234. Sorry github is very hard to follow through all the comments and with context for large PR 😐. 

```
This was a bug in the test, earlier, the random seed was not being used and the bad_signer's was actually equal to the good signer with only the address being different. This test used to pass because we used to filter out bad addresses earlier but since we don't have addressed anymore, this test was failing. The fix is to generate a radom signer whose address and private key is different from the good signer.
```